src/module/functions/mod.rs

```diff
@@ -418,6 +428,124 @@ impl Module {
 
         Ok(())
     }
+
+    /// Retrieve the ID for the first exported memory.
+    ///
+    /// This method does not work in contexts with [multi-memory enabled](https://github.com/WebAssembly/multi-memory),
+    /// and will error if more than one memory is present.
+    pub fn get_memory_id(&self) -> Result<MemoryId> {
+        if self.memories.len() > 1 {
+            bail!("multiple memories unsupported")
+        }
+
+        self.memories
+            .iter()
+            .next()
+            .map(Memory::id)
+            .context("module does not export a memory")
+    }
+
+    /// Replace a single exported function with the result of the provided builder function.
+    ///
+    /// When called, if `builder` produces a None value, the function in question will be
+    /// replaced with a stub that does nothing (more precisely, a function with an unreachable body).
+    ///
+    /// This function returns the function ID of the *new* function
+    pub fn replace_exported_func_by_id<F>(
```

Comment:
Perhaps just `replace_exported_func` instead of `by_id` for a little brevity? (and for imports)

Reply:
Yup, I'm definitely good with that -- will update

- [ ] rename `replace_exported_func_by_id` to `replace_exported_func`